gatekeeper/client/client.go

```diff
@@ -29,20 +32,46 @@ type Client struct {
 }
 
 // NewClient returns a new client to a Gatekeeper host that can bootstrap this machine
-func NewClient(host string) *Client {
+func NewClient(host, caFile string) (*Client, error) {
 	if !strings.HasSuffix(host, "/") {
 		host += "/"
 	}
+
+	caPool, err := x509.SystemCertPool()
+	if err != nil {
```

Comment:
> Also, thinking about this more, I think this would be fatal if a user didn't specify the --ca flag?

Not sure here. There are a few cases I see:
1. No `--key` and `--cert` given on `gatekeeper` cmd will start the Gatekeeper without SSL. Maybe they should be required too (`--ca` would probably need them to be required to be consistent).
2. `--key` and `--cert` specify a certificate that is signed by the system bundle (or an intermediate cert), then `--ca` wouldn't be required.

Thinking on what OpenShift does, I think (1) makes sense. In OpenShift, they generate a CA cert, and then use that as a basis of the certificates in the system. That way, there's no reliance on having a certain CA be trusted or not in your system. That should force Gatekeeper over SSL in all cases (which is probably the right thing)